File: Simultaneous2Unknown.vb
Public Class Simultaneous2Unknown

    Dim x_coefficients() As Double
    Dim y_coefficients() As Double
    Dim equal() As Double
    Dim eliminator(1, 1) As Double
    Dim x_variable As Double
    Dim y_variable As Double

    ' A constructor
    Public Sub _init_(equations)
        x_coefficients = equations("x")
        y_coefficients = equations("y")
        equal = equations("eq")
    End Sub

    ' Returns a list Of the result
    Public Function solveSimultaneous() As Double()
        ' eliminate y
        ' STEP 2:
        eliminator(0, 0) = y_coefficients(1) * x_coefficients(0)
        eliminator(0, 1) = y_coefficients(1) * equal(0)
        ' STEP 3:
        eliminator(1, 0) = y_coefficients(0) * x_coefficients(1)
        eliminator(1, 1) = y_coefficients(0) * equal(1)

        Try
            ' STEPS 4, 5:
            x_variable = (eliminator(0, 1) - eliminator(1, 1)) / (eliminator(0, 0) - eliminator(1, 0))
            ' STEP 6:
            y_variable = (equal(0) - x_coefficients(0) * x_variable) / y_coefficients(0)

            Return {x_variable, y_variable}

        Catch ex As DivideByZeroException
            Throw ex
            Return Nothing
        End Try
    End Function

End Class

File: Simultaneous2Unknown.py
# A class
class Sim2Unknown:

    # A constructor
    def __init__(self, equations):
        self.x_coefficients	= equations['x']
        self.y_coefficients	= equations['y']
        self.equals             = equations['eq']

        self.eliminator = [[[], []], [[], []]]

        
    # Returns a list of the result
    def solveSimultaneous(self):
        # eliminate y
		# STEP 2:
        self.eliminator[0][0] = self.y_coefficients[1] * self.x_coefficients[0]
        self.eliminator[0][1] = self.y_coefficients[1] * self.equals[0]
		# STEP 3:
        self.eliminator[1][0] = self.y_coefficients[0] * self.x_coefficients[1]
        self.eliminator[1][1] = self.y_coefficients[0] * self.equals[1]

        try:
			# STEPS 4, 5:
            self.x_variable = (self.eliminator[0][1] - self.eliminator[1][1]) / (self.eliminator[0][0] - self.eliminator[1][0])
			# STEP 6:
            self.y_variable = (self.equals[0] - self.x_coefficients[0] * self.x_variable) / self.y_coefficients[0]

            return [self.x_variable, self.y_variable]

        except:
            raise ValueError(None)
